browser-extension/mv2/src/background/background.js

```diff
@@ -264,13 +269,50 @@ BG.Methods.modifyHeaders = function (originalHeaders, headersTarget, details) {
     ruleType,
     rulePairs,
     rulePair,
-    isRuleApplied = false,
+    hasModifiedHeaders = false,
```

Comment:
`hasModifiedHeaders` look confusing. What is the difference between isRuleApplied and hasModifiedHeaders

Reply:
`isRuleApplied` implies that the rule has been applied instead `hasModifiedHeaders` doesn't connect rule to the modification (as in this case). Thats why the variable name was changed

Reply:
hasModifiedHeaders forces me to think in what case you want to modify headers when rule isn't applied. We should think about a more meaningful name

Reply:
Maybe `isHeadersModified` ?

Reply:
But we should still keep `isRuleApplied` variable intact